microservices/CFDP/lib/cfdp_pdu/cfdp_pdu_metadata.rb

```diff
@@ -92,16 +98,24 @@ def define_metadata_pdu_contents
     return s, s2
   end
 
-  def build_metadata_pdu_contents(source_entity:, closure_requested:, checksum_type:, file_size:, source_file_name: nil, destination_file_name: nil, options: [])
+  def build_metadata_pdu_contents(destination_entity:, segmentation_control:, source_entity:, closure_requested:, checksum_type:, file_size:, source_file_name: nil, destination_file_name: nil, options: [])
+    version = destination_entity['protocol_version_number']
     s, s2 = define_metadata_pdu_contents()
-    s.write("RESERVED", 0)
-    if closure_requested
-      s.write("CLOSURE_REQUESTED", closure_requested)
+    if version >= 1
+      s.write("SEGMENTATION_CONTROL", 0) # Always 0 in version 1
+      if closure_requested
+        s.write("CLOSURE_REQUESTED", closure_requested)
+      else
+        s.write("CLOSURE_REQUESTED", source_entity['transaction_closure_requested'])
+      end
+      s.write("CHECKSUM_TYPE", checksum_type)
     else
-      s.write("CLOSURE_REQUESTED", source_entity['transaction_closure_requested'])
+      s.write("SEGMENTATION_CONTROL", segmentation_control)
+      s.write("CLOSURE_REQUESTED", 0)
+      s.write("CHECKSUM_TYPE", 0)
     end
     s.write("RESERVED2", 0)
-    s.write("CHECKSUM_TYPE", checksum_type)
```

Comment:
This change doesn't make sense to me unless it was previous a bug. Changes appear to be:
```
RESERVED            -> SEGMENTATION_CONTROL
CLOUSURE_REQUESTED  -> CLOUSURE_REQUESTED
RESERVED2           -> CHECKSUM_TYPE
CHECKSUM_TYPE       -> RESERVED2
```
Seems like the last 2 items swapped

Reply:
The ordering here doesn't matter.